src/Compiler/pars.fsy

```diff
@@ -4462,6 +4462,30 @@ declExpr:
        let trivia: SynExprLetOrUseBangTrivia = { LetOrUseBangKeyword = rhs parseState 1 ; EqualsRange = Some mEquals }
        SynExpr.LetOrUseBang(spBind, ($1 = "use"), true, $2, $4, $7, $8, m, trivia) }
 
+  | BINDER atomicPatternLongIdent COLON typeWithTypeConstraints EQUALS typedSequentialExprBlock IN opt_OBLOCKSEP moreBinders typedSequentialExprBlock %prec expr_let
+     { // Handle type annotations on simple identifier patterns in let!/use! bindings
+       let spBind = DebugPointAtBinding.Yes(rhs2 parseState 1 7)
+       let vis, lidwd = $2
+       let pat = mkSynPatMaybeVar lidwd vis (rhs parseState 2)
+       let pat = SynPat.Typed(pat, $4, unionRanges pat.Range $4.Range)
+       let mEquals = rhs parseState 5
+       let m = unionRanges (rhs parseState 1) $10.Range
+       let trivia: SynExprLetOrUseBangTrivia = { LetOrUseBangKeyword = rhs parseState 1 ; EqualsRange = Some mEquals }
+       SynExpr.LetOrUseBang(spBind, ($1 = "use"), true, pat, $6, $9, $10, m, trivia) }
+
+  | OBINDER atomicPatternLongIdent COLON typeWithTypeConstraints EQUALS typedSequentialExprBlock hardwhiteDefnBindingsTerminator opt_OBLOCKSEP moreBinders typedSequentialExprBlock %prec expr_let
```

Comment:
This is the associated offside-sensitive version of the new type annotation rule. e.g. in cases where `let!` or `use!`  bindings are used outside of the CE context. This is invalid code but we will want the tokens to be parsed.

```fsharp
let example=
    // No braces, indentation defines scope
    let! x = async { return 1 }
    let! y = async { return 2 }
    return x + y
```